avaframe/in4Region/splitInputs.py

```diff
@@ -0,0 +1,296 @@
+    """Clip the DEM around each release scenario's centerpoint and move it to respective folders."""
+    # Read input DEM
+    demData = ascUtils.readRaster(inputDEM)
+    header = demData['header']
+    raster = demData['rasterData']
+
+    cellsize = header["cellsize"]
+    xOrigin = header["xllcenter"]
+    yOrigin = header["yllcenter"]
+
+    # Find centerpoint
+    for entry in folderList:
+        folderName = entry['folderName']
+        for properties in entry['properties']:
+            geometry = entry['geometries'][0] #first geometry in list
+            center = geometry.centroid
+
+            # Calculate bounding box indices for clipping
+            bufferSize = float(cfg['GENERAL']['bufferSize'])
+            xmin, ymin, xmax, ymax = (
+                center.x - bufferSize,
+                center.y - bufferSize,
+                center.x + bufferSize,
+                center.y + bufferSize,
+            )
+
+        # Convert bounding box to grid indices
+        colStart = max(0, int((xmin - xOrigin) / cellsize))
+        colEnd = min(header["ncols"], int((xmax - xOrigin) / cellsize))
+        # Calculate row indices
+        rowStart = max(0, int((yOrigin + header["nrows"] * cellsize - ymax) / cellsize))
+        rowEnd = min(header["nrows"], int((yOrigin + header["nrows"] * cellsize - ymin) / cellsize))
+
+        # Flip rows for bottom-left origin
+        rowStart, rowEnd = header["nrows"] - rowEnd, header["nrows"] - rowStart
+
+        # Validate grid indices
+        if colStart >= colEnd or rowStart >= rowEnd: # e.g. if the bounding box falls completely outside the DEM
+            log.warning(f"Invalid clipping bounds for '{folderName}'. Skipping.")
+            continue
+
+        # Clip DEM section
+        clippedDEM = raster[rowStart:rowEnd, colStart:colEnd]
+        log.debug(f"Clipped DEM for '{folderName}' with '{len(clippedDEM)}' rows and '{len(clippedDEM[0])}' columns")
+
+        # Prepare header for clipped DEM
+        clippedDEMHeader = {
+            "ncols": colEnd - colStart,
+            "nrows": rowEnd - rowStart,
+            "xllcenter": xOrigin + colStart * cellsize,
+            "yllcenter": yOrigin + rowStart * cellsize,
+            "cellsize": cellsize,
+            "nodata_value": header["nodata_value"],
+        }
+
+        # Define output path
+        demOutPath = outputDir / folderName / "Inputs" / f"{folderName}_DEM.asc"
+
+        # Save clipped DEM
+        try:
+            ascUtils.writeResultToAsc(clippedDEMHeader, clippedDEM, demOutPath, flip=True)
+            log.info(f"Clipped DEM saved to '{demOutPath}'.")
+        except Exception as e:
+            log.error(f"Failed to save clipped DEM for '{folderName}': {e}")
+
+    return demOutPath
+
+def splitByScenarios(folderList, outputDir):
+    """Use the scenario attribute to separate by scenarios"""
+    # Read the split shapefiles
+    for folder in folderList:
```

Comment:
as discussed, this needs to be simplified / made easier to be read. Let me know if you need help with this